CONTRIBUTING.md

```diff
@@ -2,88 +2,156 @@
 
+## Getting Started
+
+### Prerequisites
+
+-   **Rust:** This project requires Rust **version 1.84**. We recommend using `rustup` to manage your Rust installation. You can install `rustup` by following the instructions at [https://rustup.rs/](https://rustup.rs/). After installing `rustup`, ensure you have the correct version:
+
+    ```bash
+    rustup toolchain install 1.84
+    rustup default 1.84
+    ```
+
+-   **Docker and Docker Compose:** Many components of Aura are run within Docker containers. Install Docker and Docker Compose from [https://docs.docker.com/get-docker/](https://docs.docker.com/get-docker/).
+-   **System Libraries:** You may need to install additional system libraries depending on your operating system. (We should add specific instructions for common OSes here). Specifically, you'll need `protobuf-compiler`:
+    ```bash
+    sudo apt-get update && sudo apt-get install -y protobuf-compiler # Example for Debian/Ubuntu
+    ```
+-   **Git:** You'll need Git to clone the repository and manage your changes.
+
+### Setup
+
+1.  **Clone the Repository:**
+
+    ```bash
+    git clone <repository_url>
+    cd <repository_name>
+    ```
+
+2.  **Build the Project:**
+
+    ```bash
+    cargo build
+    ```
+
+    This will download dependencies and build the project. The first build may take a while.
+
+3.  **Run Unit Tests:**
+
+    ```bash
+    cargo test
+    ```
+
+4.  **Run Integration Tests:**
+
+    ```bash
+    cargo test -F integration_tests
+    ```
+
+    Note: Integration tests may require specific environment variables or setup. See the `README.md` and `docker-compose.yaml` for details. The integration tests use a local PostgreSQL instance, as configured in the `rust.yml` CI workflow.
+
+### IDE Setup (Recommended)
+
+We recommend using VS Code with the Rust Analyzer extension for development. Install the extension from the VS Code marketplace.
+
 ## GitFlow Workflow
 
 We follow the GitFlow branching model for our development process. This provides a robust framework for managing larger projects.
 
 ### Main Branches
 
-- `main` - Production-ready code. This branch contains the latest released version.
-- `develop` - Main development branch. Features and fixes are merged here before release.
+-   `main` - Production-ready code. This branch contains the latest released version.
+-   `develop` - Main development branch. Features and fixes are merged here before release.
 
 ### Supporting Branches
 
-1. **Feature Branches**
-   - Branch from: `develop`
-   - Merge back into: `develop`
-   - Naming convention: `feature/description-of-feature`
-   - Example: `feature/add-compression-support`
+1.  **Feature Branches**
+    -   Branch from: `develop`
+    -   Merge back into: `develop`
+    -   Naming convention: `feature/MTG-<ticket_number>-description-of-feature` (Include the ticket number!)
+    -   Example: `feature/MTG-123-add-compression-support`
```

Comment:
i'd add a case where a willing contributor does not have the ticket number, and as such it can be omitted